### PR TITLE
Create compiler for React pages

### PR DESCRIPTION
**Pull request by matikrk**
Instead of using html is possible create component page using react (JSX)
in component folder create folder 'pages' and put jsx files
normal component jsx files shouldn't be in 'pages' folder 
added small examples for buttons

### CODE REVIEW

**Comment by vergilius** [comment]
please add info to README with your description ;)

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (kdzwinel) [inline .babelrc]: do we need that file? Doesn't webpack set babel's config?

File: src/components/buttons/pages/buttonsReact.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ButtonPrimary from '../ButtonPrimary.jsx'
+
+const Buttons = () => {
+    return (
+        <div>
+            <ButtonPrimary type='dark'>jakis text !!!!!</ButtonPrimary>
```
Comment (burakukula) [inline src/components/buttons/pages/buttonsReact.jsx]: Copy in english please.

File: src/components/buttons/ButtonPrimary.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const ButtonPrimary = ({disabled, type, children, ...props}) => {
+    const btnClass = classNames('sg-button-primary', {
```
Comment (vergilius) [inline src/components/buttons/ButtonPrimary.jsx]: looks weird, don't we have single tab spacing? (its 2 spaces in files above)

File: package.json [hunk 4]
```diff
@@ -38,5 +45,10 @@
     "postversion": "git push && git push --tags",
     "watch": "gulp build && gulp watch & npm start",
     "test": "npm run scss-lint -s && npm run scss-unused-variables -s"
+  },
+  "dependencies": {
+    "classnames": "^2.2.5",
+    "react": "^15.5.4",
+    "react-dom": "^15.5.4"
   }
```
Comment (kdzwinel) [inline package.json]: why not devDependencies?

File: src/components/buttons/pages/buttonsReact2.jsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ButtonPrimary from '../ButtonPrimary.jsx'
+
+const Buttons = () => {
+    return (
+        <div>
+            <ButtonPrimary type='dark'>inny text</ButtonPrimary>
```
Comment (burakukula) [inline src/components/buttons/pages/buttonsReact2.jsx]: @matikrk same here.

File: src/components/buttons/ButtonPrimary.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const ButtonPrimary = ({disabled, type, children, ...props}) => {
```
Comment (vergilius) [inline src/components/buttons/ButtonPrimary.jsx]: shouldn't we immediately add callback for clicking behaviour?

File: src/components/buttons/ButtonPrimary.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    return <button {...props} disabled={disabled}  className={btnClass}>{children}</button>;
+};
+
+ButtonPrimary.propTypes = {
+    children: PropTypes.node.isRequired,
+    disabled: PropTypes.bool,
+    type: PropTypes.oneOf(['alt', 'dark'])
+};
+
+export default ButtonPrimary;
```
Comment (kdzwinel) [inline src/components/buttons/ButtonPrimary.jsx]: @aju why not 'Button' with 'type=primary/secondary' ?

File: src/components/buttons/pages/buttonsReact.jsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ButtonPrimary from '../ButtonPrimary.jsx'
```
Comment (vergilius) [inline src/components/buttons/pages/buttonsReact.jsx]: can we counfigure loader to skip extension?

File: .babelrc (hunk 1 above)
Comment (vergilius) [inline .babelrc]: we can inline it in webpack 👍

File: src/components/buttons/pages/buttonsReact.jsx (hunk 2 above)
Comment (kdzwinel) [inline src/components/buttons/pages/buttonsReact.jsx]: English please 🇬🇧 Also, make sure that styleguide looks exactly the same after this PR.

File: scripts/tasks/build-react-pages.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        file.contents = new Buffer(htmlPage);
+        cb(null, file);
+
+      }))
+      .pipe(rename(function(path) {
+        path.extname = ".html"
+      }))
+      .pipe(gulp.dest(docsOutputPath));
+  }
+};
```
Comment (kdzwinel) [inline scripts/tasks/build-react-pages.js]: I have no idea what's happening 😅 @aju you are our only hope to review this

File: package.json (hunk 4 above)
Comment (matikrk) [inline package.json]: React and react-dom is used in components and development tools so I will move this to devDep. classnames is use only in components so shouldn't moved

File: .babelrc (hunk 1 above)
Comment (matikrk) [inline .babelrc]: .babelrc define babel configuration for whole project and it's known method. We use .babelrc also in diffrent projects ex. in bulkimport. So IMHO we should use this file for configure babel

Comment (aju) [inline .babelrc]: lets keep it here easier to track changes

File: scripts/tasks/build-react-pages.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        const fileNameWithExtension = pathArray.pop();
+        const relativePath = pathArray.join('/');
+        const jsPath = plugins.path.join(consts.VERSIONED_DIST, 'docs', relativePath);
+
+        const config = Object.assign({}, coreConfig, {
+          entry: {
+            [fileNameWithExtension]: file.path
+          },
+          output: {
+            filename: "[name]",
```
Comment (vergilius) [inline scripts/tasks/build-react-pages.js]: single quotes please

File: scripts/tasks/build-react-pages.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        const path = file.path.replace(consts.SRC, consts.VERSIONED_DIST + '/docs');
+        const ReactPageClass = require(path).default;
+        const htmlPage = ReactDOMServer.renderToStaticMarkup(React.createElement(ReactPageClass));
+
+        file.contents = new Buffer(htmlPage);
+        cb(null, file);
+
+      }))
+      .pipe(rename(function(path) {
+        path.extname = ".html"
```
Comment (vergilius) [inline scripts/tasks/build-react-pages.js]: single quotes

File: src/components/buttons/pages/buttonsReact2.jsx [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ButtonPrimary from '../ButtonPrimary'
```
Comment (vergilius) [inline src/components/buttons/pages/buttonsReact2.jsx]: ';'

File: package.json [hunk 13]
```diff
@@ -38,5 +47,8 @@
     "postversion": "git push && git push --tags",
     "watch": "gulp build && gulp watch & npm start",
     "test": "npm run scss-lint -s && npm run scss-unused-variables -s"
+  },
+  "dependencies": {
+    "classnames": "^2.2.5"
```
Comment (aju) [inline package.json]: move it to dev dependencies

File: src/components/buttons/pages/buttonsReact.jsx [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ButtonPrimary from '../ButtonPrimary'
```
Comment (vergilius) [inline src/components/buttons/pages/buttonsReact.jsx]: ';'

File: src/components/buttons/pages/buttonsReact2.jsx (hunk 12 above)
Comment (vergilius) [inline src/components/buttons/pages/buttonsReact2.jsx]: you can skip fixing it until you add linter

File: scripts/tasks/build-react-pages.js [hunk 15]
```diff
@@ -0,0 +1,76 @@
+const through = require('through2');
+const fs = require('fs');
+const webpack = require('webpack');
+const React = require('react');
+const ReactDOMServer = require('react-dom/server');
+const rename = require('gulp-rename');
+
+const nodeModules = {};
+fs.readdirSync('node_modules')
```
Comment (aju) [inline scripts/tasks/build-react-pages.js]: check if there is no package to do it

File: scripts/tasks/build-react-pages.js [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+            console.error(err);
+            return;
+          }
+          cb(null, file)
+        });
+
+
+      }))
+      .pipe(through.obj((file, enc, cb) => {
+        const path = file.path.replace(consts.SRC, consts.VERSIONED_DIST + '/docs');
```
Comment (aju) [inline scripts/tasks/build-react-pages.js]: make it a functions so flow is clear

File: src/components/buttons/ButtonPrimary.jsx (hunk 6 above)
Comment (aju) [inline src/components/buttons/ButtonPrimary.jsx]: ...props solves it

File: src/components/buttons/pages/buttonsReact.jsx (hunk 8 above)
Comment (matikrk) [inline src/components/buttons/pages/buttonsReact.jsx]: done

File: src/components/buttons/pages/buttonsReact.jsx (hunk 2 above)
Comment (matikrk) [inline src/components/buttons/pages/buttonsReact.jsx]: done

File: src/components/buttons/ButtonPrimary.jsx (hunk 7 above)
Comment (kdzwinel) [inline src/components/buttons/ButtonPrimary.jsx]: @aju ping

File: src/components/buttons/pages/buttonsReact.jsx (hunk 2 above)
Comment (kdzwinel) [inline src/components/buttons/pages/buttonsReact.jsx]: kk

**Comment by aju** [comment]
@matikrk remember to bumb version

**Pull request merged**